test/e2e/storage/vsphere/vsphere_volume_node_delete.go

```diff
@@ -0,0 +1,148 @@
+
+		totalNodes := len(nodeList.Items)
+
+		node1 := nodeList.Items[0]
+
+		govMoMiClient, err := vsphere.GetgovmomiClient(nil)
+		Expect(err).NotTo(HaveOccurred())
+
+		finder := find.NewFinder(govMoMiClient.Client, true)
+		ctx, _ := context.WithCancel(context.Background())
+
+		vmPath := filepath.Join(workingDir, string(node1.ObjectMeta.Name))
+		vm, err := finder.VirtualMachine(ctx, vmPath)
+		Expect(err).NotTo(HaveOccurred())
+
+		// Find .vmx file path to be used to re register the node
+		var nodeVM mo.VirtualMachine
+		err = vm.Properties(ctx, vm.Reference(), []string{"config.files"}, &nodeVM)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(nodeVM.Config).NotTo(BeNil())
+
+		vmFolder, err := finder.FolderOrDefault(ctx, workingDir)
+		Expect(err).NotTo(HaveOccurred())
+		host, err  := vm.HostSystem(ctx)
+		Expect(err).NotTo(HaveOccurred())
+		rpool ,err := vm.ResourcePool(ctx)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Power off the node: %v", node1.ObjectMeta.Name))
+		_, err = vm.PowerOff(ctx)
+		Expect(err).NotTo(HaveOccurred())
+		err = vm.WaitForPowerState(ctx, vimtypes.VirtualMachinePowerStatePoweredOff)
+		Expect(err).NotTo(HaveOccurred(), "Unable to power off the node")
+
+		By(fmt.Sprintf("Unregister the node: %v", node1.ObjectMeta.Name))
+		err = vm.Unregister(ctx)
+		Expect(err).NotTo(HaveOccurred(), "Unable to unregister the node")
+
+		// Ready nodes should be 1 less
+		verifyReadyNodes(f.ClientSet, totalNodes - 1)
+
+		By(fmt.Sprintf("Register the node: %v", node1.ObjectMeta.Name))
+		registerTask, err := vmFolder.RegisterVM(ctx, nodeVM.Config.Files.VmPathName, node1.ObjectMeta.Name, false, rpool, host)
+		Expect(err).NotTo(HaveOccurred())
+		err = registerTask.Wait(ctx)
+		Expect(err).NotTo(HaveOccurred())
+
+		vm, err = finder.VirtualMachine(ctx, vmPath)
+		Expect(err).NotTo(HaveOccurred())
+
+		By(fmt.Sprintf("Power on the node: %v", node1.ObjectMeta.Name))
+		vm.PowerOn(ctx)
+		err = vm.WaitForPowerState(ctx, vimtypes.VirtualMachinePowerStatePoweredOn)
+		Expect(err).NotTo(HaveOccurred(), "Unable to power on the node")
+
+		// Ready nodes should be equal to earlier count
+		verifyReadyNodes(f.ClientSet, totalNodes)
+
+
+		// Sanity test that pod provisioning works
+		scParameters := make(map[string]string)
+		storagePolicy := os.Getenv("VSPHERE_SPBM_GOLD_POLICY")
+		Expect(storagePolicy).NotTo(BeEmpty(), "Please set VSPHERE_SPBM_GOLD_POLICY system environment")
+		scParameters[SpbmStoragePolicy] = storagePolicy
+		invokeValidPolicyTest(f, client, namespace, scParameters)
+	})
+})
+
+// verify ready status of nodes upto 1 minute
+func verifyReadyNodes(client clientset.Interface, expectedNodes int){
+	numNodes := 0
+	for i := 0; i < 31; i++ {
+		nodeList := framework.GetReadySchedulableNodesOrDie(client)
+		Expect(nodeList.Items).NotTo(BeEmpty(), "Unable to find ready and schedulable Node")
+		numNodes = len(nodeList.Items)
+		if numNodes == expectedNodes {
```

Comment:
This check might not always suffice, especially in face of bug. We should also check for the specific node to be missing or present in the list, along with number.